src/components/Inputs/RadioButtons/RadioButtons.js

```diff
@@ -0,0 +1,293 @@
+
+  // These will appear if RadioButtonGroup is used with redux-form:
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func,
+  onDragStart: PropTypes.func,
+  onDrop: PropTypes.func,
+  onFocus: PropTypes.func,
+}
+
+export function RadioButtonGroup({
+  options,
+  onChange,
+  formChangeHandler,
+  allCaps = true,
+  name = `radio-button-group-${uuidv4()}`,
+  initialValue = undefined,
+  currentValue,
+  currentError,
+  formTouched,
+  disabled,
+  validator,
+  required,
+  labelCopy,
+  ...rest
+}) {
+  let initialSelected
+  if (currentValue || typeof currentValue === 'boolean') {
+    initialSelected = currentValue
+  } else if (initialValue || typeof initialValue === 'boolean') {
+    initialSelected = initialValue
+  }
+
+  const [selectedValue, setSelectedValue] = useState(initialSelected)
+  const [isAnswered, setIsAnswered] = useState(false)
+  const resolvedValidator = validator ? validator : () => ''
+  const [getError, setError, , validate] = useErrorMessage(resolvedValidator)
+
+  const validationRadio = () => {
+    let errorMessage = validate(selectedValue)
+    errorMessage = errorMessage.length ? errorMessage : ''
+    setError(errorMessage)
+    if (formChangeHandler) {
+      // Update form with the new value and a falsy error message
+      formChangeHandler(selectedValue, errorMessage)
+    }
+  }
+  useEffect(() => {
+    const isSelectedValue = typeof selectedValue !== 'undefined'
+    if (onChange && isSelectedValue) {
+      onChange({ value: selectedValue, isAnswered })
+      validationRadio()
+    }
+  }, [selectedValue, isAnswered])
+
+  function onClickHandler(value, clickHandler) {
+    return (evt) => {
+      setSelectedValue(value)
+      if (!isAnswered) {
+        setIsAnswered(true)
+      }
+      return clickHandler && clickHandler(evt)
+    }
+  }
+
+  let hasChecked = false
+  const allRadioValues = []
+  const finalOptions = options.map((o) => {
+    const { value, onClick: passedHandler } = o
+    allRadioValues.push(value)
+    const checked = value === selectedValue
+    // Keeps track of whether or not we have any radio checked
+    if (checked) {
+      hasChecked = true
+    }
+    const onClick = onClickHandler(value, passedHandler)
+    return { ...o, name, checked, onClick, onChange }
+  })
+
+  const handleBlur = () => {
+    validationRadio()
+  }
+
+  // This key handling essentially allows us to comply with the spec. See their example:
+  // https://www.w3.org/TR/wai-aria-practices/examples/radio/radio-1/js/radioGroup.js
+  // And Jared Palmer's React version which he obviously referenced w3's example:
+  // https://github.com/palmerhq/radio-group/blob/master/src/index.tsx#L92
+  const handleKeyDown = React.useCallback(
+    (event) => {
+      const last = allRadioValues.length - 1
+      const index = allRadioValues.findIndex((i) => i == selectedValue)
+      event.persist()
+      var flag = false
+      function setPrevious() {
+        if (index === 0) {
```

Comment:
out of curiosity, as a JS noob, if we closed over `flag` and added `flag=true` here and in `setNext`, would this actually work?

Reply:
Funny, this whole section was very much inspired by the referenced works, and your questions helped me to track down some things that needed change.

The `var` will always get "hoisted" to top of it's scope e.g. function in this case. This is fine here, but I've switched it to `let`. Now, if we were in a "block", `let` will work like you intuit from other languages (it'll be limited to the scope of that block, whereas `var` will just get hoisted and potentially confuse the hell out of all of us :) 

But yeah, I think since it's defined in outer most part of the function it sort of doesn't matter between `var` and `let`, as you mentioned, it'd be closed over and visible from any inner functions. That said, happy we were able to remove the `var` :)

Reply:
Lol, thanks for the thorough explanation! Appreciate it! I'm still somewhat confused, I don't want to limit the scope to the block, I want the same scope it currently has, but I wanted to be able to access it from the closure. In any case, not a big deal.

Reply:
@alexthecarp If you mean outside the arrow function, I think that would work too.